### PR TITLE
MGDOBR-952: add router canonical hostname to shard table

### PR DESCRIPTION
**Pull request by r00ta**
[MGDOBR-952](https://issues.redhat.com/browse/MGDOBR-952)

Draft as I still have to show the team the general idea behind all the upcoming changes

## PR owner checklist

Please make sure that your PR meets the following requirements:

- [ ] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/main/kogito-build/kogito-ide-config).  
  *Run `mvn clean verify -DskipTests` so that imports are correctly set.*
- [ ] Your commit messages are clear and reference the JIRA issue e.g: "[MGDOBR-1] - $clear_explanation_of_what_you_did"
- [ ] The layers in the `kustomize` folder have been updated accordingly.
- [ ] All new functionality is tested
- [ ] Pull Request title is properly formatted: `MGDOBR-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

## PR reviewer(s) checklist

- [ ] Check if code and Github action workflows have been modified and if the modification is safe
- [ ] If the modification is safe, add the `safe to test` label

<details>
<summary>
How to trigger pipelines and use the bots:
</summary>

- <b>Run the end to end pipeline</b>  
  Annotate the pull request with the label: `safe to test`. If you want to run the pipeline again, remove and add it again.

- <b>Rebase the pull request</b>  
  Comment with: `/rebase`.
- <b>Deploy BOT</b>
  Comment with `/deploy <dev,stable> when the PR has been merged to deploy to a target environment.

</details>

### CODE REVIEW

**Comment by r00ta** [comment]
I'll add the router canonical hostname values in the dev/stable tables manually after I deploy this change to the envs.

File: manager/src/main/resources/db/migration/0.1/V0.1.16__add_shard_router_hostname.sql [hunk 1]
```diff
@@ -0,0 +1,2 @@
+alter table SHARD
+    add column "router_canonical_hostname" varchar(255);
```
Comment (danielezonca) [inline manager/src/main/resources/db/migration/0.1/V0.1.16__add_shard_router_hostname.sql:2]: How do you expect to migrate the existing environments? 
Is this covered by a future PR?

@robpblake 
I think we might decide to expose a `registerShard` endpoint to do the wiring and provide similar information. Wdyt?

**Review by danielezonca — APPROVED** [review]

File: manager/src/main/resources/db/migration/0.1/V0.1.16__add_shard_router_hostname.sql (hunk 1 above)
Comment (r00ta) [reply manager/src/main/resources/db/migration/0.1/V0.1.16__add_shard_router_hostname.sql:2]: as we have only one shard for dev and one shard for stable, I'll login into the cluster and update the database tables accordingly. 

We still have to design/implement all the components to terraform a shard - I'd expect this column will be populated during that process when a new shard will be registered using the admin api/whatever

Comment (robpblake) [reply manager/src/main/resources/db/migration/0.1/V0.1.16__add_shard_router_hostname.sql:2]: @r00ta I think we should automate the setting the value of the `router_canonical_hostname` in all environments. In particular in `app-interface`, we don't have the ability to log in to the DB and insert data by hand.

The column should also have the `not null` constraint as a Shard without a `router_canonical_hostname` entry is not valid.

I'd suggest we follow the approach that we've taken previously so something like:

```
quarkus.flyway.placeholders.shard-router-canonical-hostname=${SHARD_ROUTER_CANONICAL_HOSTNAME}
```
in `application.properties` and then have the migration reference the placeholder.
We can then update each environment configuration (including `app-interface`) to update the data and then set the column to have the `not null` constraint afterwards.

cheers,
Rob

**Comment by r00ta** [comment]
> @r00ta We should find a way to automate the setting of the `router_canonical_hostname` field in all environments. I've made a suggestion that aligns with the pattern we've used to set the `id` of the shard.
> 
> cheers, Rob

Could you take another look now?

**Comment by r00ta** [comment]
@msmagnanijr could you create a new secret `SHARD_ROUTER_CANONICAL_HOSTNAME` with the value of the router canonical hostname of the fleet shard in app-interface? Thanks

File: kustomize/overlays/dev/secrets/event-bridge-manager-secrets.yaml [hunk 2]
```diff
@@ -32,6 +32,7 @@ spec:
...
+    SHARD_ROUTER_CANONICAL_HOSTNAME: AgB3vCPSCHRIfKamjNN+NGipW9lNKfHWUuRJMsuoqzDdvIX1tMoKBQ43svhQ1KBWmHFNkslxYijBHPJoVxrogn1R5m+GKpJpYYWY6Rl7Ben7Oj4mOXz+w6i2H0TNxh6ovGAPfFqyfa9DkqpxQXNaalzTFeYa63HTsAhmCCM6g3KOEMS6PsoooO0Sk328r9wzJzn8eDU1tAxTaPOnMpHgvFDVghK0AVSxf6ilxY/YQ7K6/IpxtmKNMRdNNCAel2c/hXPDLk+BduqcPZPuLfgASDmIUFTgwx+Pcwi0hQtMVGiilwOP4m/HbN8v2cWGWUnm8BKkYIH5lP4EA1/oI8JhlnVDHZmSTNe5jzeBcQHFhQvz0mZF9G3WOqPy+PgiREdkQhq3ruZWJjERMKVr3vwvGB20wk2Wc3gOmYkKY6KJDOWFw9yDrtKXkcrY5kTFai4Nov8jSM68O2C/HvJGapOnMoKH9uO0WECia2PfZGBSfn5EYrmUQXZwpE74lDQeZ1CxtgWuSvUcm5j7Sl8XA3E9aFjEBWWGpArQEYvCLmxtL9TJamaZEL3TTWI52sR58thVP6f2Qe6LQRk8kVS2SPZUPzrPtjI4e58iWo4U3JLM2OIggCSoWKy1C78IKpl+fSjpzJy3+nwavNJP3reedb5UB+Jo+lCUO4Vh0eUwcWB3AL5Red+YraMorSCXYXdP/m/R2xKo3N0EqL3NNgKrJUirtanNUsDUVZMPLZgXeWumSG5LJMaEV7o2GKt1s7WBWdzevCGXT+1BBPQ6tYDIO8s=
```
Comment (jiripetrlik) [inline kustomize/overlays/dev/secrets/event-bridge-manager-secrets.yaml:35]: Are these properties encoded somehow? It sounds to me like a quite strange hostname. Can't we use something simple?

File: manager/src/main/resources/application.properties [hunk 3]
```diff
@@ -38,7 +38,8 @@ quarkus.flyway.migrate-at-start=true
 quarkus.flyway.validate-on-migrate=true
 quarkus.flyway.ignore-missing-migrations=true
 quarkus.flyway.create-schemas=true
-quarkus.flyway.placeholders.shard-id=${SHARD_ID}
+quarkus.flyway.placeholders.shard-id=${SHARD_ID:150b23e9-0b34-45b7-a91a-5ec388d03a1d}
```
Comment (jiripetrlik) [inline manager/src/main/resources/application.properties:41]: Is there any reason to start to provide default SHARD-ID here?

File: kustomize/overlays/dev/secrets/event-bridge-manager-secrets.yaml (hunk 2 above)
Comment (r00ta) [reply kustomize/overlays/dev/secrets/event-bridge-manager-secrets.yaml:35]: These are secrets. You can read more about how we handle secrets in `dev` and `stable` here https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/tree/main/kustomize#sealed-secrets

File: manager/src/main/resources/application.properties (hunk 3 above)
Comment (r00ta) [reply manager/src/main/resources/application.properties]: As it is not mandatory I've removed it

**Review by robpblake — APPROVED** [review]

**Review by jiripetrlik — APPROVED** [review]
@r00ta 
Thank you for explanation. Looks good to me. But it will need rebase.

**Comment by msmagnanijr** [comment]
> @msmagnanijr could you create a new secret `SHARD_ROUTER_CANONICAL_HOSTNAME` with the value of the router canonical hostname of the fleet shard in app-interface? Thanks

Hey @r00ta I think I missed this comment. I'll do that today and let you know.

**Pull request merged**